pkg/controller/internal/transportadapter/transportadapter_translator.go

```diff
@@ -242,6 +254,86 @@ func (t *transportAdapterTranslator) createSecretEnvFrom(
 	return envFrom
 }
 
+// createVolumes creates volumes from the MCPServer deployment spec
+func (t *transportAdapterTranslator) createVolumes(
+	deployment v1alpha1.MCPServerDeployment,
+) []corev1.Volume {
+	volumes := make([]corev1.Volume, 0)
+
+	// Add volumes from SecretRefs
+	for _, secretRef := range deployment.SecretRefs {
+		if secretRef.Name == "" {
+			continue
+		}
+		volumes = append(volumes, corev1.Volume{
+			Name: fmt.Sprintf("secret-%s", secretRef.Name),
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: secretRef.Name,
+				},
+			},
+		})
+	}
```

Comment:
I don't think we need to do this the secrets are already added to the container via the `EnvFrom` stanza.